### ISSUE TITLE
Request and response logging of the unserialized Java types

### ISSUE THREAD

**Issue by lsuski** [square/retrofit#378]
Hi,
I'd like to log only request and response body along with http status code and execution time (no headers, urls and methods names). Is there possibility to make log level mechanism more flexible? For example use some interface or simple class with methods like isLogStatus(), isLogBody() etc. and ovveride it for own needs.
Thanks

**Comment by hosamaly** [comment]
It looks like 2.0.0-beta1 removed logging support altogether in commit 40bfa340bb back in January. Any idea when it might return?

**Comment by JakeWharton** [comment]
Nope. Maybe never. Use OkHttp interceptors in the mean time.

**Comment by hosamaly** [comment]
Thanks. FWIW, here is an example for a logging interceptor for OkHttp:
https://github.com/square/okhttp/wiki/Interceptors

**Comment by vitaliyistomov** [comment]
Any chances to have default supported by Square logging interceptor implementation for OkHttp with logging of the bodies of the request and response like it was in Retrofit 1.x? I believe every second person would be interested in having that instead of tweaking the snippet from https://github.com/square/okhttp/wiki/Interceptors to log full request/response for debugging every time.

**Comment by cemo** [comment]
Is this already solved by https://github.com/square/okhttp/pull/1874?
```
OkHttpClient client = new OkHttpClient();
HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
logging.setLevel(Level.BASIC);
client.interceptors().add(logging);
```

**Comment by JakeWharton** [comment]
No. That's logging at the HTTP layer which I mostly don't care about.
Retrofit needs logging at the application layer using your types, not HTTP,
for the representation.

**Comment by JakeWharton** [comment]
Might be time for an event listener

https://publicobject.com/2022/05/01/eventlisteners-are-good/
